### ISSUE TITLE
Various UI/UX Improvement.

### ISSUE THREAD

**Issue by HarshilPatel007** [zaproxy/zaproxy#6851]
**Is your feature request related to a problem? Please describe.**
1) I like to use "Full Layout" but, It's very frustrating that I've to click "Sites Tab" every time whenever I want to see a new URLs or endpoints. I just want to see a "Sites Tab" every time when I'm working on/in another Tab(s).
2) Make "Tabs" re-arrangement possible. example. : I like to use "Requester(Add-on) Tab" besides "Request & Response Tab" or maybe say, adjust relevant Tabs according to my needs and project scope/priority.
3) It's very very frustrating to go back and forth between "Search" for searching a string in "Request & Response Tab", "Fuzzer Tab".

**Describe the solution you'd like**
1) Just enable the "Sites Tab" like, "Expand Sites Tab" in "Full Layout" mode.
2) Make "Tabs" re-arrangement possible. just enable the drag&drop to re-arrange "Tabs".
3) Add "Search Bar" or features like "Search Tab" within the "Request & Response Tab", "Fuzzer Tab" for the easy accessibility.
(note : we can merge functions/features of "Highlighter Add-on" for enhancement)

**Would you like to help implementing this feature?**
I'll help in UI/UX side things like, designing or make draft of UI/UX improvements.

**Comment by psiinon** [comment]
Help with the UI/UX designs always appreciated :) A wire frame model with some explanations would be very helpful.
FYI you do know that you can right click and "Find..." in the Request/Response tabs? That just seaches in those tabs however.

**Comment by HarshilPatel007** [comment]
> Help with the UI/UX designs always appreciated :) A wire frame model with some explanations would be very helpful. FYI you do know that you can right click and "Find..." in the Request/Response tabs? That just seaches in those tabs however.

@psiinon , yes I know that we can do "Find..." but, I think Regex is not supported there..!!! ( maybe I'm wrong )
but, I think whole functionality of "search tab" would be better combined with "Highlighter Add-on". this'll add an edge to the "Request & Response Tab", "Fuzzer Tab".

yes, I'll make a mock-ups / draft designs for UI/UX improvements later and will discuss with dev. team about it.

**Comment by HarshilPatel007** [comment]
@psiinon @thc202 ,
I've created a draft design for a "Full Layout". please give your feedback.
![image892](https://user-images.githubusercontent.com/16424866/136502132-d78544d9-a347-4b54-af6e-d59b1a3936c3.png)

1) "Full Layout" enabled/selected.
2) "Sites tree"
3) Main area.
4) Toggle buttons.

advantage of this design is,
1) users don't have to switch back & forth between "Sites Tab" and other tabs.
2) users have a choice to toggle "Sites Tab" if they want a full screen experience.
3) users can focus on main working tabs (Sites Tab & Other Tab they're working on) instead of other non-used tabs. this'll help them to up their productivity and also assist them visually as well.

will post other drafts for "Searching" a when I've time.

**Comment by HarshilPatel007** [comment]
@psiinon , do still consider this idea?

**Comment by psiinon** [comment]
As I mentioend above, I'm fine with it.
Its not something I would work on, but I'm happy for anyone else to work on it.
